### DISCUSSION TITLE
Need to reset the page Index in table's state when search occurs

### DISCUSSION THREAD

**Discussion by Waqqars987** [TanStack/table#2864] [Q&A] [answered]
Hi,

I have a requirement where I need to reset the page Index in the table's state to 0 whenever a search happens. I am using manual server sided pagination.
Please tell me how to achieve this.

**Comment by umarluqman** [comment, marked as answer]
You can take a look at this 
https://codesandbox.io/s/react-table-with-global-states-pagination-index-ny4kj?file=/src/Listing.js:4367-4454

Basically, it uses the global states (page index from an API) instead of the table local state when the user is not using the built-in pagination function.
